llama_stack/providers/adapters/inference/ollama/ollama.py

```diff
@@ -234,5 +234,30 @@ async def embeddings(
         self,
         model: str,
         contents: List[InterleavedTextMedia],
+        sampling_params: Optional[SamplingParams] = SamplingParams(),
+        truncate: Optional[bool] = True,
+        logprobs: Optional[LogProbConfig] = None,
     ) -> EmbeddingsResponse:
-        raise NotImplementedError()
+
+        request = EmbeddingRequest(
+            model=model,
+            contents=contents,
+            sampling_params=sampling_params,
+            truncate=truncate,
+            logprobs=logprobs,
+        )
+        return await self._embeddings(request)
+
+    async def _embeddings(self, request: EmbeddingRequest) -> EmbeddingsResponse:
+        params = self._get_params_for_embeddings(request)
+        r = await self.client.embed(**params)
+        return EmbeddingsResponse(embeddings=r["embeddings"])
+
+    def _get_params_for_embeddings(self, request: EmbeddingRequest) -> dict:
+        sampling_options = get_sampling_options(request)
+        return {
+            "model": OLLAMA_SUPPORTED_MODELS[request.model],
```

Comment:
I don't think OLLAMA_SUPPORTED_MODELS works here -- embedding models are different in nature and there has to be a separate list.

Reply:
It some how worked for Llama3.1-8B-Instruct, from my testing, but I did realize later that it doesnt quite work in the case of fireworks and inference. I will update the PR with a separate list

Reply:
Llama3.1-8B-Instruct is not an embedding model actually.